addons/point_of_sale/static/src/js/models.js

```diff
@@ -1057,72 +1057,67 @@ class Product extends PosModel {
     get_price(pricelist, quantity, price_extra){
         var self = this;
         var date = moment();
-
-        // In case of nested pricelists, it is necessary that all pricelists are made available in
-        // the POS. Display a basic alert to the user in this case.
-        if (!pricelist) {
-            alert(_t(
-                'An error occurred when loading product prices. ' +
-                'Make sure all pricelists are available in the POS.'
-            ));
-        }
```

Comment:
Yes, I know, but I want first see what comportment we want in the POS as soon as we have a pricelist available. Should it be by default, as it's now (again ^^) the case for the SO, or should I let it as is where we could have advanced pricelists but fallback on the product price + POS currency when none are selected by default?

Case one: if default, check it's there and so reintroduced the warning
Case two: we can get rid of the warning, or maybe just first check if there is a Pricelist in the default one or not before

For now, it's case two that I choose, but I know we'll probably go to case one. There is another TODO about that in the POS, there are multiple places that must be changed for case one. Since case one is more similar to master than in this PR, it'll be easy to catch should I need to change.

Reply:
The solution is probably to give a param to know if we enter this method recursively from a pricelist rule based on a pricelist, or if we are in the first call of the price computation (in which case, having no pricelist is fine).